### DISCUSSION TITLE
Match only file in root directory of repo

### DISCUSSION THREAD

**Discussion by AKorezin** [renovatebot/renovate#9083] [Request Help] [answered]
I have a project with files
```
/test/docker-compose.yml
/docker-compose.yml
```
I want to match only file `/docker-compose.yml`. And I'm failed to do that.

I tried to use this config
```json
{
  "extends": [
    "config:base",
  ],
  "pinDigests": true,
  "packageRules": [
    {
      "matchFiles": ["docker-compose.yml"],
      "pinDigests": false
    }
  ],
}
```
But I've got an error
```json
{
  "validationMessage": "packageRules: Each packageRule must contain at least one selector (matchPackageRules, matchPaths, matchLanguages, matchBaseBranches, matchManagers, matchDatasources, matchDepTypes, matchPackageNames, matchPackagePatterns, excludePackageNames, excludePackagePatterns, matchCurrentVersion, matchSourceUrlPrefixes, matchUpdateTypes). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all."
}
```

So I tried to use this config
```json
{
  "extends": [
    "config:base",
  ],
  "pinDigests": true,
  "packageRules": [
    {
      "matchPaths": ["/docker-compose.yml"],
      "pinDigests": false
    }
  ],
}
```
But it didn't match `/docker-compose.yml`.

How can I pin digests in `/test/docker-compose.yml` but not in `/docker-compose.yml`?

**Comment by viceice** [comment, marked as answer]
checkout [matchFiles](https://docs.renovatebot.com/configuration-options/#matchfiles)

**Reply by AKorezin** [reply]
Already did that. Got an error.
Is it usable only in combination with `matchPaths`?

**Reply by viceice** [reply]
Please show the full rule you want to have. I do not understand why you like to mix `matchPaths` with `matchFiles`

**Reply by AKorezin** [reply]
In my project I have files
```
/docker-compose.yml
/first-project/docker-compose.yml
/second-project/docker-compose.yml
/another-project/main/Dockerfile
/another-project/another/Dockerfile
```
I want to pin digests in all `docker-compose.yml` and `Dockerfile` files except `/docker-compose.yml`
I expect that this config will work.
```json
{
  "extends": [
    "config:base",
  ],
  "pinDigests": true,
  "packageRules": [
    {
      "matchFiles": ["docker-compose.yml"],
      "pinDigests": false
    }
  ],
}
```
But it won't work. Instead I've got an error.

**Reply by viceice** [reply]
Yes, found a bug, see #9084. Workaround:

```json
{
  "extends": [
    "config:base",
  ],
  "pinDigests": true,
  "packageRules": [
    {
      "matchPaths": ["+(docker-compose.yml)"],
      "pinDigests": false
    }
  ],
}
```

see [matchPaths](https://docs.renovatebot.com/configuration-options/#matchpaths) for explanation

**Reply by AKorezin** [reply]
But it wouldn't match
```
/first-project/docker-compose.yml
/second-project/docker-compose.yml
```
So this files will remain without digests.

**Reply by viceice** [reply]
?? that rule should pin all digests excerpt `/docker-compose.yml`. Did you tried it?

**Reply by AKorezin** [reply]
Sorry my fault.
Placed file in `/test/` directory in project.

**Reply by viceice** [reply]
yes, test folders are excluded by `config:base`, can be overriden by:

![image](https://user-images.githubusercontent.com/1798109/110804492-8264c900-8280-11eb-997f-52eb745a4eda.png)


https://github.com/visualon/renovate-config/blob/9806bc3e89337a1fcef3507e025f4508c3e0bcfb/shared.json#L18-L26

**Comment by dmt0** [comment]
This is what worked for me:
`"matchPaths": ["+(docker-compose.yml)"],`
